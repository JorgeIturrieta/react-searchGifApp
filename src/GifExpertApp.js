
import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid';


export const GifExpertApp = ( {defaultCategories = []}) => {


    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h1>GifExpertApp</h1>
            <hr />
            <AddCategory setCategories={setCategories} />
            <ol>
                {categories.map((category) => (
                    <GifGrid
                     key= {category}
                     category = { category }/>
                )
                   
                )
                }
            </ol>
        </>
    )
}
