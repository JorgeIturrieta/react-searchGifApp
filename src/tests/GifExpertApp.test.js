import { shallow } from "enzyme" ;
import {GifExpertApp} from '../GifExpertApp'

describe('Tests de componente <GifExpertApp />', () => {
    
})

test('Debe ser igual al snapshot', () => {
    
    const wrapper = shallow(<GifExpertApp />);
      expect(wrapper).toMatchSnapshot();
});

test('Debe de mostrar una lista de categorias', () => {
    const categories = ['Dragon Ball','Samurai x'];
    const wrapper = shallow(<GifExpertApp defaultCategories = {categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);

})

