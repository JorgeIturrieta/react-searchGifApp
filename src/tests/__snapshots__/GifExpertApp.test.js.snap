// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Samurai x"
      key="Samurai x"
    />
  </ol>
</Fragment>
`;

exports[`Debe ser igual al snapshot 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
